src/core/server/server.api.md

```diff
@@ -1525,6 +1531,32 @@ export interface SavedObjectsClientWrapperOptions {
     request: KibanaRequest;
 }
 
+// @public (undocumented)
+export interface SavedObjectsComplexFieldMapping {
+    // (undocumented)
+    dynamic?: string;
+    // (undocumented)
+    properties: SavedObjectsMappingProperties;
+    // (undocumented)
+    type?: string;
+}
+
+// @public (undocumented)
+export interface SavedObjectsCoreFieldMapping {
+    // (undocumented)
+    enabled?: boolean;
```

Comment:
Can we link to elasticsearch docs describing the format of the mappings? https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-types.html 
Is it possible to add proper typings for in the future?

Reply:
> Can we link to elasticsearch docs describing the format of the mappings?

Yea (thanks for the link). However these type are wrongly defined as `public` (as I picked the commits from my 'mappings' PR), and are not ATM exposed in any API. I would like to keep all documentation in the next PR where we expose `registerType`. Would declaring them back to `internal` for now be acceptable?

> Is it possible to add proper typings for in the future?

We do want that. However I would really like to avoid redefining all direct ES types, and I think these types are not defined in the ES library we are using. Maybe this could be resolved when using the new `es` node library?

Reply:
> However I would really like to avoid redefining all direct ES types, and I think these types are not defined in the ES library we are using. Maybe this could be resolved when using the new es node library?

I didn't find them in `elasticsearch-js` either. @delvedor are there any plans to add them?

Reply:
Hello! The new js client does not have type definitions for Elasticsearch's request and response bodies, only for URL and query parameters.
It does offer a solution for fixing this through generics, see the [typescript documentation](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/typescript.html).

Reply:
Added link to the documentation. The generic does not answer our needs though, as in our part of the SO code, we have no way to be aware of the actual SO type we are working on.

We will have to improve the typings on our side I guess, but this should be done in another PR (We need improvement on pretty much everything in the `src/core/server/saved_objects` subfolders...)

Reply:
> We need improvement on pretty much everything in the src/core/server/saved_objects subfolders...

Would you mind creating a list of follow-up tasks?

Reply:
Sure, will add this as a condition to merge this PR.

Reply:
Created https://github.com/elastic/kibana/issues/56856